docs/configuring-entities.md

```diff
@@ -0,0 +1,197 @@
+
+```
+service1/
+├── docs/
+│   └── index.md
+├── mkdocs.yaml
+docs/
+├── index.md
+mkdocs.yaml
+```
+
+Depending on which setup you will chose the configuration of the workflow will be different.
+
+### Using TechDocs workflow template
+
+A template can be found [here][9], after the template has been coppiced uncomment this part:
+
+```yaml
+on:
+  push:
+    branches:
+      - main
+      - master
+    paths:
+      - "**/docs/**"
+      - "mkdocs.yaml"
+      - "mkdocs.yml"
+```
+
+since it will allow the workflow to run on any changes made to the documentation.
+
+Next up define the steps for the workflow. Each step has to reference [this][11] action which is used to activate the build and publish process. A code snipped below explains how to use this action when your documentation is placed in the root of your directory:
+
+```yaml
+- name: Send dispatch for service-catalog
+  uses: operate-first/service-catalog/.github/actions/docs-dispatch@main
+  with:
+    # Root of the repository
+    docs_path: '.'
+    # Name of your entity in the catalog
+    entity_name: 'service-catalog'
+    # Type of your entity in the catalog
+    entity_kind: 'Component'
+    # This can always be ${{ github.repository }}
+    repository: ${{ github.repository }}
+    # GitHub token that requires metadata:read and contents:read&write permissions to service-catalog repository
+    token: ${{ secrets.SESHETA_TOKEN }}
+```
+
+If the documentation is not located in the root of the repository, a filter definition is required using [glob expresions][15]:
+
+```yaml
+- name: Send dispatch for service2
+  uses: operate-first/service-catalog/.github/actions/docs-dispatch@main
+  with:
+    # Glob expression, filtering where your documentation root is
+    filter: 'service2/**'
+    # Directory path to your documentation folder relative to the repository root
+    docs_path: 'service2'
+    entity_name: 'service2'
+    entity_kind: 'Component'
+    repository: ${{ github.repository }}
+    token: ${{ secrets.SESHETA_TOKEN }}
+```
+
+<!--Mermaid diagram showing techdocs architecture somewhere here -->
+[Documentation reference][13]
+
+## GitHub Insights/Security Insights
+
+For the insights plugins to work it is required to put `github.com/project-slug` annotation to your components definition. The value of this annotation is a GitHub project slug which consists of the organization/user and the repository name, see the example below.
+
+```yaml
+annotations:
+    github.com/project-slug: operate-first/service-catalog
+```
+
+As long as the repository is public `GitHub Insights` will be shown, for one exception. If the repository is private and the GitHub account that is used for the login and has the permission for the repository, insights will be also shown. This applies to `Security Insights` too, if the GitHub account has the correct permissions to view the repository security insights, the plugin will show them.
```

Comment:
This is not very understandable... 

> As long as the repository is public `GitHub Insights` will be shown, for one exception.

What exception?


Please rewrite this in a more approachable manner. Something like:

```suggestion
Both of these plugins (`GitHub Insights` and `Security Insights`) interface directly to GitHub on front-end. Therefore some information visibility is subjective to each user. In general following rules apply:

GitHub Insights plugin:
1. Repository is private
  a. User has access to the repository -> Content is visible to the user
  b. User doesn't have access to the repository -> Content is not visible to the user
2. Repository is public -> Content is visible to the user

Security Insights:
1. User is granted permissions to view security advisories -> Security insights is visible to the user
```

Reply:
Its a lot better to split it up into actual rules, gonna try to utilize this more when writing more documentation.

Resolved in the latest commit.